### PR TITLE
Add tests for promote_rule, widen, mod, bswap, ~

### PR DESCRIPTION
**Pull request by ScottPJones**
This adds tests for various parts of int.jl that were not covered previously.
Except for two sections where the code is different for 32-bit vs. 64-bit, all of the current code
in int.jl should now be covered.

### CODE REVIEW

**Comment by ScottPJones** [comment]
@kshyatt @mbauman @dpsanders Some more coverage, passes unit tests

**Comment by ScottPJones** [comment]
Bump - hopefully this will get int.jl fully covered (as well as simplifying the code in both base/int.jl and test/int.jl), dropped 69 lines while adding tests! :grinning:

**Comment by dpsanders** [comment]
Looks good to me!

File: base/int.jl [hunk 1]
```diff
@@ -203,46 +198,28 @@ rem{T<:Integer}(x::T, ::Type{T}) = x
...
+    for T in SignedIntTypes
+        @eval convert(::Type{$T}, x::$Tf) = box($T,checked_fptosi($T,unbox($Tf,x)))
     end
-end
-
-for to in (UInt8, UInt16, UInt32, UInt64)
-    @eval begin
-        convert(::Type{$to}, x::Float32) = box($to,checked_fptoui($to,unbox(Float32,x)))
-        convert(::Type{$to}, x::Float64) = box($to,checked_fptoui($to,unbox(Float64,x)))
+    for T in UnsignedIntTypes
```
Comment (yuyichao) [inline base/int.jl]: IIUC, the definition for `UInt128`/`Int128` is override below.

Comment (ScottPJones) [inline base/int.jl]: Yes, that was a typo, should have been Signed64Types and Unsigned64Types, instead of SignedIntTypes/UnsignedIntTypes.  Now fixed.

**Comment by ScottPJones** [comment]
@yuyichao now updated, passing tests

**Comment by tkelman** [comment]
said it before, will say it again: -1 on moving 32-bit-only code into a separate file

**Comment by ScottPJones** [comment]
Ok, merged back in

**Comment by ScottPJones** [comment]
@tkelman Merged 32-bit stuff in, has passed tests, any more changes required?

**Comment by ViralBShah** [comment]
I presume we want to hold this until 0.4 branches.

**Comment by ScottPJones** [comment]
Could just the 2nd commit be merged in to 0.4? (the part that is simply adding coverage tests)

**Comment by ViralBShah** [comment]
Let's split out the tests into a separate PR, and we can merge that.

**Comment by ScottPJones** [comment]
Ok, I'm in the air on my way home from Belgium, I'll do it as soon as I've got a decent connection at home.

**Comment by ScottPJones** [comment]
Ok, I'm in the air on my way home from Belgium, I'll do it as soon as I've got a decent connection at home.

**Comment by ScottPJones** [comment]
@ViralBShah Except for unrelated OOM killer problem (in parallel test), this passes.  Anything else or can it be merged?
@tkelman What is going on with the OOM killer problem again?

**Comment by ScottPJones** [comment]
Bump, these tests should be able to go in now, pass all tests

**Comment by ScottPJones** [comment]
@jakebolewski Could you kindly take a look at this one?  Again, it's just coverage testing, filling in the holes.

**Pull request merged**